media_driver/agnostic/gen11/codec/kernel_free/build.py

```diff
@@ -0,0 +1,199 @@
+import os,shutil
+import sys
+
+import struct
+import os
+from sys import platform
+
+class GenSource2Bin(object):
+    def __init__(self, component='codec'):
+        self.fromdir    = './Binary'
+
+    def iga_compile(self):
+        for filename in os.listdir('./Source'):
+            fp = os.path.join('./Source', filename)
+            fileext = os.path.splitext(filename)[1]
+            if os.path.isfile(fp) and 'cpp' in fileext:
+                command  = 'cmc /c /Qxcm -Qxcm_jit_target=gen11lp '
+                command += fp
+                command += ' -mCM_emit_common_isa  -mCM_no_input_reorder -mCM_jit_option="-nocompaction" '
+                os.system(command)
+
+                print("Compiled %s" % fp);
+
+    def deleteFile(self):
+        for filename in os.listdir('./'):
+            fp = os.path.join('./', filename)
+            fileext = os.path.splitext(filename)[1]
+            if os.path.isfile(fp) and 'asm' in fileext:
+                os.remove(fp)
+
+            if os.path.isfile(fp) and 'dat' in fileext:
+                os.remove(fp)
+
+            if os.path.isfile(fp) and 'visaasm' in fileext:
+                os.remove(fp)
+
+            if os.path.isfile(fp) and 'isa' in fileext:
+                os.remove(fp)
+
+    def join(self):
+        isExists=os.path.exists("Binary")
+        if not isExists:
+            os.mkdir("Binary")
+
+        shutil.copy('downscale_kernel_genx_0.dat', 'Binary/DS4x_Frame.krn')
+        shutil.copy('downscale_kernel_genx_1.dat', 'Binary/DS2x_Frame.krn')
+        shutil.copy('hme_kernel_genx_0.dat', 'Binary/HME_P.krn')
+        shutil.copy('hme_kernel_genx_1.dat', 'Binary/HME_B.krn')
+        shutil.copy('hme_kernel_genx_2.dat', 'Binary/HME_VDENC.krn')
+        shutil.copy('downscale_convert_kernel_genx_0.dat', 'Binary/DS_Convert.krn')
+
+        output = open('commonkernel.krn', 'wb')
+
+        bytes  = struct.pack('i', 6)
+        output.write(bytes)
+
+        count  = 64
+        bytes  = struct.pack('i', count)
+        output.write(bytes)
+
+        filepath = os.path.join('./Binary', 'DS4x_Frame.krn')
```

Comment:
This block from line 61 to line 67 is a common pattern and should be extracted to the separate function.